### PR TITLE
Fix SQL Object + Positional parameter + No -parameters compile flag

### PR DESCRIPTION
**Pull request by qualidafial**
Addresses #843.

@arteam or @stevenschlansker: Could you please review?

### CODE REVIEW

File: noparameters/pom.xml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+~   distributed under the License is distributed on an "AS IS" BASIS,
+~   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+~   See the License for the specific language governing permissions and
+~   limitations under the License.
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
```
Comment (arteam) [inline noparameters/pom.xml]: It's really pity there is no "noparameters" flag for the java compiler, so we are forced to copypast the maven compiler plugin configuration in all modules.

File: noparameters/src/test/java/org/jdbi/v3/noparameters/TestSqlObjectNoParameterNames.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+    @Test
+    public void namedParameterWithNoParameterNamesCompiledOrSupplied() throws Exception {
+        assertThatThrownBy(() -> h.attach(Dao.class).getByNamedId(1))
+                .isInstanceOf(UnableToExecuteStatementException.class)
+                .hasMessageContaining("no named parameter matches 'id'");
+    }
+
+    @RegisterBeanMapper(Something.class)
+    public interface Dao {
```
Comment (arteam) [inline noparameters/src/test/java/org/jdbi/v3/noparameters/TestSqlObjectNoParameterNames.java]: I think we could add a test for the error message of a missed name for the `@BindList` annotation, too.

File: core/pom.xml [hunk 3]
```diff
@@ -218,6 +218,16 @@
                 </executions>
             </plugin>
             -->
+
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
```
Comment (stevenschlansker) [inline core/pom.xml]: Could we avoid the copypasta by activating this in the parent with a profile?  The profile could be active by default and suppressed by a `.javac-no-parameters` flag file or the like.

Comment (qualidafial) [inline core/pom.xml]: I'll try it.

Comment (qualidafial) [inline core/pom.xml]: That worked. I used a property that's only defined in the noparameters pom. Thanks

File: noparameters/src/test/java/org/jdbi/v3/noparameters/TestSqlObjectNoParameterNames.java (hunk 2 above)
Comment (qualidafial) [inline noparameters/src/test/java/org/jdbi/v3/noparameters/TestSqlObjectNoParameterNames.java]: Agreed. Probably likewise for `@Define` and `@DefineList`.

File: core/pom.xml (hunk 3 above)
Comment (qualidafial) [inline core/pom.xml]: Spoke too soon. From what I can tell, the set of activated profiles is locked in at the parent pom--it doesn't look like you can change it in individual artifacts. Reinstating copypasta. _sad panda_

**Pull request merged**